aws_batch/non_host_alignment.py

```diff
@@ -88,17 +88,28 @@
 AWS_BATCH_JOB_ID = None
 
 # convenience functions
-def generate_taxid_annotated_fasta_from_m8(input_fasta_file, m8_file, output_fasta_file, annotation_prefix):
+def fuzzy_get(dictionary, raw_key, default_value):
+    value_list = [dictionary[key] for key in dictionary if raw_key in key]
```

Comment:
wouldn't this be kinda slow?

Reply:
yeah. It's only used for re-processing old samples, not new samples. I think the trouble was that old samples had annotations that made it impossible to know where the actual raw read-id started. Need to look into it again. Or suggestions welcome...